build.ps1

```diff
@@ -63,19 +63,47 @@ $LOGDIR = $ENLISTMENT_ROOT + "\bin"
 # Default to use Visual Studio 2015
 $VS14MSBUILD=$PROGRAMFILESX86 + "\MSBuild\14.0\Bin\MSBuild.exe"
 $VSTEST = $PROGRAMFILESX86 + "\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe"
-$SN = $PROGRAMFILESX86 + "\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\sn.exe"
-$SNx64 = $PROGRAMFILESX86 + "\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\x64\sn.exe"
+
+# Figure out the directory and path for SN.exe
+$SN = $null
+$SNx64 = $null
+ForEach ($directory in [System.IO.Directory]::EnumerateDirectories($PROGRAMFILESX86 + "\Microsoft SDKs\Windows"))
+{
+    ForEach($sdkVersionDirectory in [System.IO.Directory]::EnumerateDirectories($directory + "\bin"))
+    {
+        if([System.IO.File]::Exists($sdkVersionDirectory + "\sn.exe") -and [System.IO.File]::Exists($sdkVersionDirectory + "\x64\sn.exe"))
+        {
+            $SN = $sdkVersionDirectory + "\sn.exe"
+            $SNx64 = $sdkVersionDirectory + "\x64\sn.exe"
+            break
+        }
+    }
+}
 
 # Use Visual Studio 2017 compiler for .NET Core and .NET Standard. Because VS2017 has different paths for different
 # versions, we have to check for each version. Meanwhile, the dotnet CLI is required to run the .NET Core unit tests in this script.
+# Furthurmore, Visual Studio 2017 has a Preview version as well which uses Microsoft Visual Studio\Preview as path instead of \2017
```

Comment:
Not sure why you need to update build.ps1 for this task. The dev environment typically requires VS2015 and VS2017 installed. Is this the case for your local dev environment as well?

Reply:
I am running VS2017Preview version. Since we are asking people to run VS2017, dev can choose to run preview or stable version.